### ISSUE TITLE
Add InsertQuery.setSelect(Collection, Select) overload

### ISSUE THREAD

**Issue by chilliger** [jOOQ/jOOQ#9692]
While the DSL API accepts collections like this:
`create.insertInto(Table<R> into, Collection<? extends Field<?>> fields).select(Select<?> select)`

The model API does not:
`create.insertQuery(Table<R> into).setSelect(Field<?>[] fields, Select<?> select)`

Internal the array is converted to a list anyway:
```
public final void setSelect(Field<?>[] f, Select<?> s) {
        insertMaps.addFields(Arrays.asList(f));
        select = s;
    }
```

**Comment by lukaseder** [comment]
Thanks a lot for your suggestion. Yes, makes sense to overload this method, like all the others.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.13.0. Backport scheduled for 3.12.4 (#9694)
